architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,227 @@
+
+### Pooling
+
+User connections can be multiplexed over shared HBONE connections.
+This is done through standard HTTP/2 pooling.
+The pooling is keyed off the `{source identity, destination identity, destination ip}`.
+
+### Headers
+
+Ztunnel uses the following well-known headers in HBONE:
+
+| Header        | Purpose                                                                                                                                                                                                                                                 |
+|---------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `:authority`  | Required in `CONNECT`, this is the target destination                                                                                                                                                                                                   |
+| `Forwarded`   | For outgoing requests, the original source IP. Note that since we spoof IPs in most cases, this usually is the same as the actual IP seen. For incoming requests, this is used only for traffic from waypoints (which are trusted and cannot spoof IPs) |
+| `Baggage`     | (Experimental, likely to be removed) This contains metadata about the source/destination workload for telemetry purposes                                                                                                                                |
+| `Traceparent` | (Experimental) This maintains tracing information. Note this is tracing of *connections*, and is not correlated to tracing of user's own HTTP requests. However, this is useful to follow a connection across ztunnels.                                 |
+
+## Traffic routing
+
+Based on the three [redirection](#redirection) paths, the ztunnel has three primary types of traffic it handles.
+
+### Outbound
+
+Requests leaving a pod go through the "outbound" code path on port 15001.
+This is where most of Ztunnel's logic lives.
+
+For outbound traffic, we need to first determine where the traffic is destined to.
+As Ztunnel operates at L4, we only have the destination IP/port (recovered via `SO_ORIGINAL_DST`).
+This may be an IP of a Service, a Pod, or something outside the cluster.
+Ztunnel will look up the destination from the [addresses](#address-type) it is configured with.
+
+For traffic to unknown addresses, or to workloads that are not a part of the mesh, the traffic will just be passed through as is.
+To make ztunnel more transparent, the original source IP address will be spoofed.
+Additionally, `splice` will be used to make this proxying more efficient when possible.
+
+For traffic in the mesh, things are a bit more complex:
```

Comment:
nit: prefer numbered list here to indicate that this logic is in precedence order (e.g. if the destination is on our node and has a waypoint, the first bullet occurs instead of the second)